### PR TITLE
More optimiser rules for LT/GT/AND/BYTE and address cleanups 

### PR DESCRIPTION
**Pull request by axic**
Part of #6107.

### CODE REVIEW

File: libevmasm/RuleList.h [hunk 1]
```diff
@@ -146,6 +146,11 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart2(
 		{{Instruction::SHR, {0, X}}, [=]{ return X; }, false},
 		{{Instruction::SHL, {X, 0}}, [=]{ return u256(0); }, true},
 		{{Instruction::SHR, {X, 0}}, [=]{ return u256(0); }, true},
+		{{Instruction::LT, {X, 0}}, [=]{ return u256(0); }, true},
+		{{Instruction::GT, {X, 0}}, [=]() -> Pattern { return {Instruction::ISZERO, {{Instruction::ISZERO, {X}}}}; }, false},
+		{{Instruction::GT, {X, ~u256(0)}}, [=]{ return u256(0); }, true},
+		{{Instruction::AND, {{Instruction::BYTE, {X, Y}}, {u256(0xff)}}}, [=]() -> Pattern { return {Instruction::BYTE, {X, Y}}; }, true},
```
Comment (chriseth) [inline libevmasm/RuleList.h]: Why did you set the flag to true?

File: libevmasm/RuleList.h [hunk 2]
```diff
@@ -146,6 +146,11 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart2(
...
+		{{Instruction::BYTE, {X, 31}}, [=]() -> Pattern { return {Instruction::AND, {X, u256(0xff)}}; }, false}
```
Comment (chriseth) [inline libevmasm/RuleList.h]: Why did you set the flag to true?

File: libevmasm/RuleList.h (hunk 1 above)
Comment (chriseth) [reply libevmasm/RuleList.h]: Does it automatically add `and(0xff, byte(x, y))`?

File: libevmasm/RuleList.h [hunk 3]
```diff
@@ -146,6 +146,11 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart2(
 		{{Instruction::SHR, {0, X}}, [=]{ return X; }, false},
 		{{Instruction::SHL, {X, 0}}, [=]{ return u256(0); }, true},
 		{{Instruction::SHR, {X, 0}}, [=]{ return u256(0); }, true},
+		{{Instruction::LT, {X, 0}}, [=]{ return u256(0); }, true},
+		{{Instruction::GT, {X, 0}}, [=]() -> Pattern { return {Instruction::ISZERO, {{Instruction::ISZERO, {X}}}}; }, false},
```
Comment (chriseth) [inline libevmasm/RuleList.h:150]: What about `gt(0, X)`?

File: libevmasm/RuleList.h (hunk 1 above)
Comment (axic) [reply libevmasm/RuleList.h]: It should because `AND` is set to commutative.

Comment (axic) [reply libevmasm/RuleList.h]: > Why did you set the flag to true?

That was a typo.

File: libevmasm/RuleList.h (hunk 3 above)
Comment (axic) [reply libevmasm/RuleList.h:150]: I think that should be fine.

**Comment by axic** [comment]
Updated. Please review carefully before merging.

File: libevmasm/RuleList.h (hunk 1 above)
Comment (chriseth) [reply libevmasm/RuleList.h]: Ah right, at the outside it's fine.

File: libevmasm/RuleList.h (hunk 3 above)
Comment (chriseth) [reply libevmasm/RuleList.h:150]: By which rule?

Comment (chriseth) [reply libevmasm/RuleList.h:150]: Ah you added it! (Or I didn't see it before)

File: libevmasm/RuleList.h [hunk 4]
```diff
@@ -146,6 +146,12 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart2(
 		{{Instruction::SHR, {0, X}}, [=]{ return X; }, false},
 		{{Instruction::SHL, {X, 0}}, [=]{ return u256(0); }, true},
 		{{Instruction::SHR, {X, 0}}, [=]{ return u256(0); }, true},
+		{{Instruction::LT, {X, 0}}, [=]{ return u256(0); }, true},
```
Comment (chriseth) [inline libevmasm/RuleList.h:149]: What about `lt(0, x)`?

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Merging for now, can add the rule later.

**Pull request merged**